src/main/java/hu/elte/eserial/builder/CompoundBuilder.java

```diff
@@ -0,0 +1,69 @@
+package hu.elte.eserial.builder;
+
+import hu.elte.eserial.exception.EserialInstantiationException;
+import hu.elte.eserial.exception.EserialInvalidMethodException;
+import hu.elte.eserial.model.Setter;
+import hu.elte.eserial.util.MethodUtils;
+import hu.elte.eserial.util.TypeUtils;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Map;
+
+/**
+ * Builds Compound objects.
+ */
+public class CompoundBuilder extends AbstractBuilder {
+
+    /**
+     * Constructs an {@link CompoundBuilder} and sets the {@code type} in it.
+     *
+     * @param type the class to be used in the {@link AbstractBuilder#build} method
+     */
+    CompoundBuilder(Type type) {
+        super(type);
+    }
+
+    /**
+     * @param value {@inheritDoc}
+     * @param <T> {@inheritDoc}
+     * @return a compound object of the given class and initialized with the map parameter
+     */
+    @Override
+    public <T> T build(Object value) {
```

Comment:
Here are some variable names in this method:
value, clazz, that, fieldValue
Looking at the code a few days/weeks/months later, could you tell what these variables represent without looking through the whole method to find out what you do with them? I don't think so.

I'd say this is a very important aspect to writing maintainable code, and I'd advise making your variable names more descriptive.
For example how would I know what is an input and what is an output type in this method? (This applies to the `SimpleBuilder` more.)

It's okay to have loop variables called "method" because they are only used in like 2 lines of code, but "`value`" and "`that`" have a much larger scope. Please try to rename these.